api/routes.py

```diff
@@ -143,33 +158,58 @@ async def get_all_mappings():
 
 
 @app.put("/concepts/{id}/mappings", tags=["concepts", "mappings"])
-async def create_concept_and_attach_mapping(id: str, terminology_id: str, concept_name: str, text: str):
+async def create_concept_and_attach_mapping(concept_id: str,  concept_name: str, terminology_name, text: str):
     try:
-        terminology = repository.session.query(Terminology).filter(Terminology.id == terminology_id).first()
-        if not terminology:
-            raise HTTPException(status_code=404, detail=f"Terminology with id {terminology_id} not found")
-        concept = Concept(terminology=terminology, name=concept_name, id=id)
+        if not repository._terminology_exists(terminology_name):
+            raise HTTPException(status_code=404, detail=f"Terminology {terminology_name} not found")
+        result = repository.client.query.get(
+            "Terminology",
+            ["name", "_additional { id }"]
+        ).with_where({
+            "path": "name",
+            "operator": "Equal",
+            "valueText": terminology_name
+        }).do()
+        terminology_data = result["data"]["Get"]["Terminology"][0]
+        terminology_id = terminology_data["_additional"]["id"]
+        terminology = Terminology(name=terminology_name, id=terminology_id)
+        concept = Concept(terminology=terminology, pref_label=concept_name, concept_identifier=concept_id)
         repository.store(concept)
         embedding = embedding_model.get_embedding(text)
-        mapping = Mapping(concept=concept, text=text, embedding=embedding)
+        model_name = embedding_model.get_model_name()
+        mapping = Mapping(concept=concept, text=text, embedding=embedding, sentence_embedder=model_name)
         repository.store(mapping)
-        return {"message": f"Concept {id} created or updated successfully"}
+        return {"message": f"Concept {concept_id} created successfully"}
     except Exception as e:
-        raise HTTPException(status_code=400, detail=f"Failed to create or update concept: {str(e)}")
+        raise HTTPException(status_code=400, detail=f"Failed to create concept: {str(e)}")
 
 
 @app.put("/mappings/", tags=["mappings"])
-async def create_or_update_mapping(concept_id: str, text: str):
+async def create_mapping(concept_id: str, text: str):
     try:
-        concept = repository.session.query(Concept).filter(Concept.id == concept_id).first()
-        if not concept:
+        if not repository._concept_exists(concept_id=concept_id):
             raise HTTPException(status_code=404, detail=f"Concept with id {concept_id} not found")
+        result = repository.client.query.get(
```

Comment:
this logic should be part of datastew imo

Reply:
You're right, we can have functions like `get_terminology(terminology_name: str)` and `get_concept(concept_id: str)`. Should I start implementing it?